math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java

```diff
@@ -149,11 +167,29 @@ public V removeVertex(int v) {
         return obj;
     }
 
+    private void cleanVertices(int v) {
+        vertices.remove(v);
+        removedVertices.remove(v);
+        if (removedVertices.contains(v - 1)) {
+            cleanVertices(v - 1);
+        }
+    }
+
     @Override
     public int getVertexCount() {
         return vertices.size() - removedVertices.size();
     }
 
+    @Override
+    public int getMaximumVertexIndex() {
+        for (int i = vertices.size(); i > 0; i--) {
+            if (!removedVertices.contains(i - 1)) {
+                return i - 1;
+            }
+        }
+        return -1;
+    }
```

Comment:
I'm not sure to understand what this method does:
`if (!removedVertices.contains(i - 1))` seems equivalent to `if (vertices.at(i) != null` ?

Is it really possible to have null vertices at the end of the list. The cleanVertices should avoid that situation?

Reply:
Yes, this was coded before I thought of doing `cleanVertices`, i will modify it

Reply:
At the end, I deleted this method as it is redundant with `getNodeCapacity`